apps/token-manager/app/src/components/SideBar.js

```diff
@@ -18,52 +15,17 @@ const DISTRIBUTION_COLORS = [
   '#80AEDC',
 ]
 
-const calculateStakes = (accounts, total, tokenDecimalsBase) => {
-  // Calculate the stake percantage accurate to milliether
-  const tokenMillietherBase = new BN(tokenDecimalsBase.div(new BN(1000)))
-  const maxDisplayed = DISTRIBUTION_ITEMS_MAX - 1
-  const byStake = (a, b) => b.stake - a.stake
-
-  const stakes = accounts.map(({ address, balance }) => ({
-    name: address,
-    stake:
-      // Calcaulate percentages as JS numbers
-      Math.round(
-        (balance.div(tokenMillietherBase).toNumber() /
-          total.div(tokenMillietherBase).toNumber()) *
-          100
-      ),
+const displayedStakes = (accounts, total) => {
+  const positiveAccounts = accounts.filter(({ balance }) => balance > 0)
+  return stakesPercentages(
+    positiveAccounts.map(({ balance }) => balance),
+    total,
+    DISTRIBUTION_ITEMS_MAX
+  ).map((stake, index) => ({
+    name: index === -1 ? 'Rest' : positiveAccounts[index].address,
```

Comment:
Will index ever be `-1` in a map callback?

Reply:
Oh good catch thanks! It should be [`stake.index`](https://github.com/aragon/aragon-apps/pull/432/files/0692d4d31bdc021cbf154ad9462fe25ff7cd6e42..027017c5f963aab9a77dc71aa49c03dd1d887786#diff-98ac6f8b731c6c061aa7771f0739c351R50), I’ll send an update 👍